stdlib/public/core/Array.swift

```diff
@@ -1865,3 +1865,129 @@ internal struct _ArrayAnyHashableBox<Element: Hashable>
     return true
   }
 }
+
+extension Array : Differentiable where Element : AdditiveArithmetic & Differentiable {
```

Comment:
`Element` should not be required to conform to `AdditiveArithmetic`.
```suggestion
extension Array : Differentiable where Element : Differentiable {
```